### PR TITLE
User history: record Front visits separately

### PR DESCRIPTION
**Pull request by stephanfowler**
New format in for history storage: 

```
"film":[
   "Film",          // display name
   [[2,1],[7,3]],   // Content visits (one visit two days ago, three visits seven days ago 
   [[0,1]]          // Front visits (one visit today)
],
```

IMPORTANT: history logging on R2 must be killed off before merging this.

### CODE REVIEW

**Comment by robertberry** [comment]
What's this for, yo?

**Comment by stephanfowler** [comment]
@robertberry it's to allow us to give much higher weight to front visits, thus getting rid of the whitelist, hopefully. The whitelist was added to filter out tags picked up from content visits (eg. news topics that felt wrong when they appeared in your history list.)

File: static/src/javascripts/projects/common/modules/onward/history.js [hunk 1]
```diff
@@ -277,14 +288,18 @@ define([
         return (str || '').split(',')[0];
     }
 
-    function collapseTag(t) {
-        t = t.replace(/^\/|\/$/g, '');
-        if (t.match(isEditionalisedRx)) {
-            t = t.replace(stripEditionRx, '');
+    function collapsePath(t) {
+        if (t) {
+            t = t.replace(/^\/|\/$/g, '');
```
Comment (robertberry) [inline static/src/javascripts/projects/common/modules/onward/history.js]: :mask:

**Comment by robertberry** [comment]
Bravo! :+1:

**Pull request merged**